oeps/oep-0002-bp-repo-metadata.rst

```diff
@@ -136,74 +178,29 @@ For example, in the `edx-platform`_ repo, the file might look like:
-in wide use and are provided here so that they can be used consistently.
-
-``webservice``:
-    Indicate that this repository is the root of a webservice.  eg. It can serve
-    HTTP content.
+OEPs Rationale
+--------------
 
-``library``:
-    The repository is an installable package that is reusable but doesn't run standalone.
-    eg. a pip installable pyton package
-
-``backend-service``:
-    A service that is run as a part of openedx but is not really interacted with
-    by the end-users.  eg. xqwatcher or ecommerce-worker
-
-``backend-tooling``:
-    Scripts or configuration used in conjunction with services or in support of openedx.
-
-``xblock``:
-    Indicating that a repo contains an xblock or xblock related tooling.
-
-
-Change History
-==============
-
-2019-10-29
-----------
-
-* Minor formatting and wording changes for clarity.
-
-2017-01-18
-----------
-
-* Change ``obsolete`` to ``archived`` and relax the requirement for an owner
-  if ``archived`` is True.
-
-2017-01-10
-----------
-
-* Support the ``applicable`` key in the reason dictionary.
+The design of the ``oeps`` dictionary was guided by a couple of use cases:
 
-2016-10-13
-----------
+1. Adding a new OEP that few repositories will support, initially. We shouldn't need to update all repositories to detect if they are in accordance, we should be able to assume that they aren't, or automatically detect whether they are.
+2. Repositories may have specific requirements that force them to not implement a best practice. The tools should be able to present that reasoning to anyone looking across repositories, and the reasons should be documented in the repositories themselves.
+3. As much as possible, Best Practices should be autodetected, but because they will often involve a judgement call, autodetection shouldn't be mandatory.
 
-* Move the definition of ``openedx-release`` to :doc:`OEP-10 <oep-0010>`.
+Ownership Rationale
+-------------------
 
-2016-08-24
-----------
+Keeping the granularity of **code ownership to high-level repositories** affords us:
 
-* Add documentation of the ``openedx-release``, ``track-pulls``, ``dead``,
-  and ``nick`` keys in the ``openedx.yaml`` file.
+#. Stronger collaboration on ownership with business functions since a high-level repo is directly aligned to a business need.
+#. Minimization of bookkeeping maintenance as code and owners evolve. However, we will need to ensure that all lower-level repos are still owned indirectly via high-level repos.
 
-2016-06-29
-----------
+Keeping the granularity of **human owners to individual teams** affords us:
 
-* Original publication
+#. Using existing organizational structures that have scrum processes in-place and cross-functional communication channels to follow through on their ownership responsibilities.
+#. Robust and longer-lasting owner relationships that overcome the absences and attrition of individuals.
```

Comment:
Does it actually guarantee long-lasting owner relationships, though? Until the Py3 upgrade, which required combing through openedx.yaml files, I'd regularly see things like `edx/tnl` or `edx/educator` as the owner of a repository. That was earlier this year.

I don't have a preference of team ownership vs. individual ownership. However, I do think we need one of the following, else the ownership definitions will continue to fall out-of-date and only be semi-useful.
1. Automated tooling that ensures that the owning individual is still part of the edX org, or
2. a. Automated tooling that ensures that the owning team exists, *and*
    b. Having someone who is responsible for managing & pruning edX GitHub teams, *and*
    c. Having a reliable way to get in contact with a team based on their GitHub team name (Slack channels often don't line up with teams)

Reply:
To me, this update to the docs is the first step in making ownership match what we want instead of what we need to settle for.  

I think it's success will be indicated by how quickly it is applied to our code as well as how quickly we start referring to it for planning and coordination.  For this, some tools will definitely need to be built so that we can easily consume all this data.